pkg/mysqldriver/tls.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysqldriver
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+// OpenTLS according to the passed parameters, decide whether to register tls config for mysql driver
+func OpenTLS(tlsName, mysqlCaCertAddr, mysqlClientCertAddr, mysqlClientKeyAddr string) error {
+	if tlsName == "" || mysqlCaCertAddr == "" {
+		return nil
+	}
+
+	rootCertPool := x509.NewCertPool()
+	pem, err := ioutil.ReadFile(mysqlCaCertAddr)
+	if err != nil {
+		return err
+	}
+	if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+		return fmt.Errorf("failed to append PEM")
+	}
+
+	if mysqlClientCertAddr == "" || mysqlClientKeyAddr == "" {
+		// skip client cert
+		err = mysql.RegisterTLSConfig(tlsName, &tls.Config{
+			RootCAs:            rootCertPool,
+			InsecureSkipVerify: true,
```

Comment:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/1272861/159204017-c4b268d3-7b49-4ce8-8a85-94d734b442ae.png">

I invesgated the reference, and you cannot skip verify because this verification if for server but not client.